src/modules/bundle.ts

```diff
@@ -459,7 +459,14 @@ export class BundleModule
       })
       .filter((b) => b.balance.gt(0));
     return await Promise.all(
-      ownedBalances.map(async (b) => await this.get(b.tokenId.toString())),
+      ownedBalances.map(async ({ tokenId, balance }) => {
+        const token = await this.get(tokenId.toString());
+        if (balance.gt(Number.MAX_SAFE_INTEGER.toString())) {
+          return { ...token, ownedByAddress: Number.MAX_SAFE_INTEGER };
```

Comment:
should never work with straight `number` types. `balance` is a `BigNumber` type and *should* be a `BigNumber` type for that reason. can be rendered as `toString()` in the client and can use `BigNumber` math functions to do math with

Reply:
I'm not sure exactly how I can do that though, because `ownedByAddress` is specified as a `number` instead of `BigNumber` - so don't I have to convert to a `number` to store on that field?

```
export interface BundleMetadata {
  creator: string;
  supply: BigNumber;
  metadata: NFTMetadata;
  ownedByAddress: number;
  underlyingType: UnderlyingType;
}
```

I do agree that it'd be better as a `BigNumber` but I'm not sure how else I can store it on `ownedByAddress` given the existing type there.

Reply:
I've pushed a change that makes the type of `ownedByAddress` be `BigNumber` which makes everything just work. I've made it a separate commit because I haven't run all the tests locally yet (collection.test.ts all pass) and I'm not sure whether you'll want that type being changed.

Reply:
this LGTM